raiden/tests/integration/api/rest/test_payments.py

```diff
@@ -148,6 +148,107 @@ def test_api_payments(
+        "target_address": to_checksum_address(target_address),
+        "token_address": to_checksum_address(token_address),
+        "amount": str(amount),
+        "identifier": str(identifier),
+    }
+
+    # Test a normal payment
+    request = grequests.post(
+        api_url_for(
+            api_server_test_instance,
+            "token_target_paymentresource",
+            token_address=to_checksum_address(token_address),
+            target_address=to_checksum_address(target_address),
+        ),
+        json={
+            "amount": str(amount),
+            "identifier": str(identifier),
+            "paths": [
+                {
+                    "route": [to_checksum_address(our_address), to_checksum_address(app1.address)],
+                    "address_metadata": {
+                        to_checksum_address(our_address): our_metadata,
+                        to_checksum_address(app1.address): app1.transport.address_metadata,
+                    },
+                }
+            ],
+        },
+    )
+
+    with watch_for_unlock_failures(*raiden_network):
+        response = request.send().response
+    assert_proper_response(response)
+    json_response = get_json_response(response)
+    assert_payment_secret_and_hash(json_response, payment)
+
+    # Test a payment without providing an identifier
+    payment["amount"] = "1"
+    request = grequests.post(
+        api_url_for(
+            api_server_test_instance,
+            "token_target_paymentresource",
+            token_address=to_checksum_address(token_address),
+            target_address=to_checksum_address(target_address),
+        ),
+        json={"amount": "1"},
+    )
+    with watch_for_unlock_failures(*raiden_network):
+        response = request.send().response
+    assert_proper_response(response)
+    json_response = get_json_response(response)
+    assert_payment_secret_and_hash(json_response, payment)
+
+    # Test that trying out a payment with an amount higher than what is available
+    # returns an error
+    payment["amount"] = str(deposit)
+    request = grequests.post(
+        api_url_for(
+            api_server_test_instance,
+            "token_target_paymentresource",
+            token_address=to_checksum_address(token_address),
+            target_address=to_checksum_address(target_address),
+        ),
+        json={"amount": str(deposit)},
+    )
+    response = request.send().response
+    assert_proper_response(response, status_code=HTTPStatus.CONFLICT)
+
+    # Test that querying the internal events resource works
+    limit = 5
+    request = grequests.get(
+        api_url_for(
+            api_server_test_instance, "raideninternaleventsresource", limit=limit, offset=0
+        )
+    )
+    response = request.send().response
+    assert_proper_response(response)
+    events = response.json()
+    assert len(events) == limit
+    assert all("TimestampedEvent" in event for event in events)
+
+
```

Comment:
I would like to see some tests for the unhappy paths of `paths` validation.

Reply:
✅